ballista/rust/scheduler/src/state/task_manager.rs

```diff
@@ -60,6 +59,23 @@ pub struct TaskManager<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan>
     scheduler_id: String,
     // Cache for active execution graphs curated by this scheduler
     active_job_cache: ExecutionGraphCache,
+    pending_task_queue_size: AtomicUsize,
+}
+
+impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> Clone
+    for TaskManager<T, U>
+{
+    fn clone(&self) -> Self {
+        Self {
+            state: self.state.clone(),
+            clients: self.clients.clone(),
+            session_builder: self.session_builder,
+            codec: self.codec.clone(),
+            scheduler_id: self.scheduler_id.clone(),
+            active_job_cache: self.active_job_cache.clone(),
+            pending_task_queue_size: AtomicUsize::new(self.get_pending_task_queue_size()),
```

Comment:
Is this the behaviour we want from `Clone`?

Reply:
What else behavior can we expect from `clone`?

Reply:
I believe that clone of `TaskManager` should track the same `pending_task_queue_size` as original instance, because it tracks the same jobs and scheduler state as well.

```rust
let manager_a: TaskManager<_, _> = ...;
let manager_b = manager_a.clone();


assert!(manager_a.get_pending_task_queue_size() == manager_b.get_pending_task_queue_size());
manager_a.increase_pending_queue_size(1);
// task queue sizes should be the same for clones
assert!(manager_a.get_pending_task_queue_size() == manager_b.get_pending_task_queue_size());
```

Reply:
Very good point, thank you!